modules/sdk/project-templates/project-templates-cdi-bean-portlet/src/main/resources/archetype-resources/src/main/java/portlet/__className__Portlet.java

```diff
@@ -0,0 +1,39 @@
+package ${package}.portlet;
+
+import ${package}.constants.${className}PortletKeys;
+
+import com.liferay.bean.portlet.LiferayPortletConfiguration;
+
+import javax.inject.Inject;
+
+import javax.portlet.PortletConfig;
+import javax.portlet.annotations.LocaleString;
+import javax.portlet.annotations.PortletConfiguration;
+import javax.portlet.annotations.RenderMethod;
+
+/**
+ * @author ${author}
+ */
+@PortletConfiguration(
+	portletName = ${className}PortletKeys.${className},
+	title = @LocaleString(value = ${className}PortletKeys.${className}))
+@LiferayPortletConfiguration(
+	portletName = ${className}PortletKeys.${className},
+	properties = {
+		"com.liferay.portlet.display-category=category.sample",
+		"com.liferay.portlet.instanceable=true"
+	}
+)
+public class ${className}Portlet {
+
+	@Inject
+	PortletConfig portletConfig;
+
+	@RenderMethod(
+		include = "/WEB-INF/jsp/view.jsp",
+		portletNames = {${className}PortletKeys.${className}})
+	public String doView() {
+		return "Hello from " + portletConfig.getPortletName();
```

Comment:
What does the "doView()" method do if there is a JSP?  as include?  before or after?

Reply:
Thanks a sweet annotation btw!

Reply:
To answer your question, the portlet will render the following:
```<p>Hello from Foo</p>```
```<p>Hello from view.jsp</p>```